### PR TITLE
Ladder optin

### PR DESCRIPTION
**Pull request by ahanden**
Re #185  - Adds a button to allow players to opt in or out of the ladder on the ladder page.

### CODE REVIEW

**Comment by ahanden** [comment]
I used a couple eslint-ignore statements because I was totally confounded on why I got the import/no-named-as-default error.

**Review by atompkins — COMMENTED** [review]
this isn't quite right. you can tell because callApp should only be used under /app/.
/ladder/ladderStatus should be calling /_dataAccess/daSettings and then dealing with the response
/ladder/toggle should be calling /_dataAccess/daSettings and /_dataAccess/daSettingsFlags
OptIn.svelte should be calling /ladder/ladderStatus & /ladder/toggle

_dataAccess is for accessing the API so the files should resemble the API, the only reason for it's existence is in case of fallback
/app/ is API access so is structured like the API
under /ladder/ is your code where you do what you want with the data received from the API

make sense?

is there any way we can get rid of the .then()'s in OptIn.svelte ?

**Comment by ahanden** [comment]
DeepScan doesn't seem to recognize how reduce works with a second argument.

**Comment by atompkins** [comment]
DeepScan is complaining because x !== undefined is always true. nothing can be equal to undefined, even undefined is not equal to undefined. we have /common/isUndefined for this purpose.

**Comment by ahanden** [comment]
I switched to isUndefined. Just another wat for js, I guess.

File: src/modules/_dataAccess/fallbacks/settings.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    fds[2].get('ui_preference_30') === '1',
+    fds[2].get('ui_preference_21') === '1',
+    fds[2].get('ui_preference_33') === '1',
+    fds[2].get('ui_preference_23') === '1',
+  ];
+}
+
+export default async function settings() {
+  const settingsHTML = await indexAjaxData({ cmd: 'settings' });
+  if (isUndefined(settingsHTML)) {
```
Comment (atompkins) [inline src/modules/_dataAccess/fallbacks/settings.js:59]: is this necessary?

**Comment by atompkins** [comment]
my mistake. hasError cannot return undefined. weirdly it can return an object or false.
your code should read
```if (check) { return check; }```

**Review by atompkins — CHANGES_REQUESTED** [review]

File: src/modules/ladder/OptIn.svelte [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+  let opt;
+  async function init() {
+    opt = await isOnLadder();
+  }
+  const loadPromise = init();
+
+  let togglePromise;
+
+  function toggle() {
```
Comment (atompkins) [inline src/modules/ladder/OptIn.svelte:15]: i think there should be a sendEvent in here

**Review by atompkins — APPROVED** [review]

**Pull request merged**